### PR TITLE
Multi package

### PR DESCRIPTION
**Pull request by dlthomas**
Break queryparser project into multiple packages, for eventual upload to hackage.

### CODE REVIEW

File: demo/LICENSE [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Copyright (c) 2017 David L. L. Thomas
```
Comment (lrewega) [inline demo/LICENSE]: I am guessing we require assignment. I'm sure the CLA bot will chime in.

Comment (dlthomas) [inline demo/LICENSE]: Err, yes, good catch.  Auto-generated files...

**Comment by lrewega** [comment]
I think you'll need to fixup the old commits with the correct copyright, or we could squash this PR.

**Comment by dlthomas** [comment]
I can do.

**Comment by h4v0kh3l1** [comment]
Looks good; for me it says one check is pending, but the CLA is signed, so something's inconsistent. Strange how it's for only one commit.

You have signed the CLA, yes?

**Comment by dlthomas** [comment]
I have, yeah.  I agree it's weird!

**Pull request merged**